tests/Rules.Framework.Tests/RulesEngineTests.cs

```diff
@@ -74,6 +74,52 @@ public async Task AddRuleAsync_GivenEmptyRuleDataSource_AddsRuleSuccesfully()
                 It.Is<EvaluationOptions>(eo => eo == evaluationOptions)), Times.Never());
         }
 
+        [Fact]
```

Comment:
There's no tests added here for when ThereAreRulesInDataSource. I guess that's because it is being already covered in the integration tests?